stock_picking_declared_value/models/stock_move.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2025 Nitrokey GmbH
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    sale_unit_price = fields.Float(
+        string="Unit Price",
+        digits="Product Price",
+        compute="_compute_sale_unit_price",
+        store=True,
+        help="Unit price from the sale order line",
+    )
+    discount = fields.Float(
+        string="Discount (%)",
+        digits="Discount",
+        compute="_compute_sale_unit_price",
+        store=True,
+        help="Discount from the sale order line",
+    )
+    price_subtotal = fields.Monetary(
+        string="Subtotal",
+        compute="_compute_price_subtotal",
+        store=True,
+        help="Subtotal amount for the move line",
+    )
+    currency_id = fields.Many2one(
+        related="picking_id.currency_id",
+        string="Currency",
+        readonly=True,
+    )
+
+    @api.depends("sale_line_id.price_unit", "sale_line_id.discount")
+    def _compute_sale_unit_price(self):
+        """Copy the price and discount from the sale order line"""
+        for move in self:
+            if move.sale_line_id:
+                move.sale_unit_price = move.sale_line_id.price_unit
```

Comment:
Depending on the taxes this can mean net or gross prices. If you have mixed taxes (included or excluded in price) you are adding net and gross prices and this shouldn't be intended.

The other field is called `price_subtotal` => I guess you want to ensure that you copy a net price. There is `price_reduce_taxexcl` on the `sale.order.line` but this will already include the discount.

Reply:
I prefer to use price_reduce_taxexcl so that we don't need to copy the discount field separately. Ok?

Reply:
If a discount always means that the value of the product decreased that's fine. Otherwise I would be sceptical.